builtin/game/misc_s.lua

```diff
@@ -21,7 +21,7 @@ function core.get_position_from_hash(hash)
 	local y = (hash % 65536) - 32768
 	hash  = math.floor(hash / 65536)
 	local z = (hash % 65536) - 32768
-	return vector.new(x, y, z)
+	return vector(x, y, z)
```

Comment:
…and so on, everywhere.
I'd *hope* we can guarantee built-in code isn't going to give invalid arguments, maybe expose `fast_new`?
Otherwise, I'm going to perf test `vector.new()` vs `vector()`.

Reply:
With LuaJIT, the perf is slightly worse, but this is 268,435,456 iterations so it's safe to ignore I think :P
![image](https://user-images.githubusercontent.com/27222523/190715618-c06d2041-f55e-49e3-bbdf-eb98c8ab03e0.png)
![image](https://user-images.githubusercontent.com/27222523/190716258-54c48112-a1f6-423e-bf9f-9231a1376a6f.png)

```lua
local ITERS = 2^28
minetest.register_chatcommand("test_vecdotnew", { func = function()
	local v = vector.zero()
	for i = 1, ITERS do
		v = v + vector.new(math.random(), math.random(), math.random())
	--	v = v + vector(math.random(), math.random(), math.random())
	--	v = v + { x=math.random(), y=math.random(), z=math.random() }
	end
	return true, tostring(v).." vector.new()"
end })
```

Reply:
I get about the same time for both of them. LuaJIT should be able to lift the indexing required by `vector()`/`vector.new()` out of the loop.

Reply:
Also, the argument type checks are zero cost in traces, since traces always specialize to specific types anyway.